### PR TITLE
Sound Refactor

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Sound refactor

- Updates the sound system

- 3D sounds are still a thing

- Pressure/atmospherics still impacts sounds

- Fixes a runtime with sound preference and lobby music

- Adds in support for atoms/effects playing sound
 - Yes, we can now make map elements that play sounds when stepped on

Tested out; seems to behave properly, but extensive sound testing can sometimes prove difficult without it being in an environment with lots of players.

### CODE REVIEW

**Comment by tigercat2000** [comment]
did you test it with the play sound from intercoms button? because if that works any better, it would be even better.

File: code/modules/mob/living/carbon/human/life.dm [hunk 1]
```diff
@@ -1393,13 +1393,6 @@ var/global/list/brutefireloss_overlays = list("1" = image("icon" = 'icons/mob/sc
 			if (getToxLoss() >= 45 && nutrition > 20)
 				vomit()
 
-		//0.1% chance of playing a scary sound to someone who's in complete darkness
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/life.dm]: ... this should probably be a prob(1) chance, but i don't think you should remove it

**Comment by Fox-McCloud** [comment]
Encourage people to test this out to see how it behave/plays.

Seems to definitely dampen the range of sounds.

**Comment by Fox-McCloud** [comment]
Not sure if I like this PR or not; on one hand, it's a nice buff to antags and cuts down on processing a bit, on the other--beign able to hear things around the station is interesting.

**Comment by Robertha89** [comment]
This actually reverts the sound distance to what it was iirc

**Comment by Fox-McCloud** [comment]
@Robertha89 

Not exactly: https://github.com/ParadiseSS13/Paradise/blob/990090326d163160e32549453c02762480221d2f/code/game/sound.dm that's what we had before.

**Comment by Spacemanspark** [comment]
I liked the sound system allowing us to hear decent distances.
Other than that, :+1:

**Comment by Fox-McCloud** [comment]
I don't like this PR.

**Pull request closed without merging**